src/types/base.ts

```diff
@@ -14,14 +14,14 @@
  * limitations under the License.
  */
 
-import type {CLSMetric} from './cls.js';
-import type {FCPMetric} from './fcp.js';
-import type {FIDMetric} from './fid.js';
-import type {INPMetric} from './inp.js';
-import type {LCPMetric} from './lcp.js';
-import type {TTFBMetric} from './ttfb.js';
+import type {CLSMetric, CLSMetricWithAttribution} from './cls.js';
+import type {FCPMetric, FCPMetricWithAttribution} from './fcp.js';
+import type {FIDMetric, FIDMetricWithAttribution} from './fid.js';
+import type {INPMetric, INPMetricWithAttribution} from './inp.js';
+import type {LCPMetric, LCPMetricWithAttribution} from './lcp.js';
+import type {TTFBMetric, TTFBMetricWithAttribution} from './ttfb.js';
 
-export interface Metric {
+export interface MetricBase {
```

Comment:
Given that this interface definition is used in the README documentation, my preference would be to keep the name the same.

What's the rationale for changing it?

Reply:
> What's the rationale for changing it?

To make it clearer that this is a base interface used for the properties shared between the metrics, not a type you should be encountering or declaring a value to be. The union-of-metrics type is something you may encounter, though (I meant to rename s`MetricType`/`Metric` but I forgot).

My proposal was going to be remove it from the README and have the individual metrics documented instead with the base properties inlined into each. Writing that out now, it's probably a bit verbose.

Could maybe be documented like

```ts
interface CLSMetric {
  name: 'CLS';
  entries: LayoutShift[];
}

interface FCPMetric {
  name: 'FCP';
  entries: PerformancePaintTiming[];
}

// ...
```
Then something like "All metrics share the following properties:"
```ts
interface MetricBase { // (or whatever)
  /**
   * The current value of the metric.
   */
  value: number;

  // ...current `Metric` with `name` and `entries` removed
}
```

Reply:
Actually, playing with it, I don't think the `Metric` or `MetricType` names are that important on their own, so I'm happy leaving them unchanged. I will try out that adjustment to the README, though.